openmc/trackfile.py

```diff
@@ -0,0 +1,168 @@
+from collections import namedtuple
+
+import h5py
+
+from .checkvalue import check_filetype_version
+from .source import SourceParticle, ParticleType
+
+
+ParticleTrack = namedtuple('ParticleTrack', ['particle', 'states'])
+ParticleTrack.__doc__ = """\
+Particle track information
+
+Parameters
+----------
+particle : openmc.ParticleType
+    Type of the particle
+states : numpy.ndarray
+    Structured array containing each state of the particle. The structured array
+    contains the following fields: ``r`` (position; each direction in [cm]),
+    ``u`` (direction), ``E`` (energy in [eV]), ``time`` (time in [s]), ``wgt``
+    (weight), ``cell_id`` (cell ID) , and ``material_id`` (material ID).
+
+"""
+
+_VERSION_TRACK = 3
+
+
+def _identifier(dset_name):
+    """Return (batch, gen, particle) tuple given dataset name"""
+    _, batch, gen, particle = dset_name.split('_')
+    return (int(batch), int(gen), int(particle))
+
+
+class Track:
+    """Tracks resulting from a single source particle
+
+    Parameters
+    ----------
+    dset : h5py.Dataset
+        Dataset to read track data from
+
+    Attributes
+    ----------
+    identifier : tuple
+        Tuple of (batch, generation, particle number)
+    particles : list
+        List of tuples containing (particle type, array of track states)
+    sources : list
+        List of :class:`SourceParticle` representing each primary/secondary
+        particle
+
+    """
+
+    def __init__(self, dset):
+        tracks = dset[()]
+        offsets = dset.attrs['offsets']
+        particles = dset.attrs['particles']
+        self.identifier = _identifier(dset.name)
+
+        # Construct list of track histories
+        tracks_list = []
+        for particle, start, end in zip(particles, offsets[:-1], offsets[1:]):
+            ptype = ParticleType(particle)
+            tracks_list.append(ParticleTrack(ptype, tracks[start:end]))
+        self.particles = tracks_list
+
+    def __repr__(self):
+        return f'<Track {self.identifier}: {len(self.particles)} particles>'
+
+    def plot(self, axes=None):
```

Comment:
I wonder if it is worth returning the ```plt`` object so user can customize it

Reply:
Absolutely, I've gone ahead and updated this